.github/workflows/release.yml

```diff
@@ -6,6 +6,101 @@ on:
+
+  shell-scripts:
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'release PR') }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - machine: windows-2022
+            log-dir: "/c/ProgramData/OpenTelemetry .NET AutoInstrumentation/logs"
+          - machine: ubuntu-20.04
+            log-dir: "/var/log/opentelemetry/dotnet"
+          - machine: macos-11
+            log-dir: "/var/log/opentelemetry/dotnet"
+    runs-on: ${{ matrix.machine }}
+    steps:
+      - uses: actions/setup-dotnet@v3.0.3
+        with:
+          dotnet-version: | 
+            6.0.x
+            7.0.x
+      - run: brew install coreutils
+        if: ${{ runner.os == 'macOS' }}
+      - name: Create test directory
+        run: sudo mkdir -p ${{ matrix.log-dir }} && sudo chmod a+rwx ${{ matrix.log-dir }}  # workaround before next release
+        if: ${{ runner.os != 'Windows' }}
+      - name: Test the Shell scripts from README.md
+        shell: bash
+        run: |
+          set -e
+          dotnet publish -f net7.0 -c Release ./test/test-applications/integrations/TestApplication.Smoke
+          curl -sSfL https://raw.githubusercontent.com/${{ github.repository }}/${{ github.sha }}/otel-dotnet-auto-install.sh -O
+          sh ./otel-dotnet-auto-install.sh
+            test "$(ls -A "$HOME/.otel-dotnet-auto")"
+          curl -sSfL https://raw.githubusercontent.com/${{ github.repository }}/${{ github.sha }}/instrument.sh -O
+          . ./instrument.sh
+          ./test/test-applications/integrations/TestApplication.Smoke/bin/Release/net7.0/publish/TestApplication.Smoke
+            test "$(ls -A '${{ matrix.log-dir }}' )"
+
+  shell-scripts-container:
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'release PR') }}
+    strategy:
+      fail-fast: false
+      matrix:
+        base-image: [ alpine ]
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v3.1.0
+    - name: Test the Shell scripts from README.md in Docker container
+      run: |
+        set -e
+        docker build -t mybuildimage -f "./docker/${{ matrix.base-image }}.dockerfile" .
+        docker run --mount type=bind,source="${GITHUB_WORKSPACE}",target=/project --rm mybuildimage /bin/sh -c '
+          set -e
+          dotnet publish -f net7.0 -c Release ./test/test-applications/integrations/TestApplication.Smoke
+          curl -sSfL https://raw.githubusercontent.com/${{ github.repository }}/${{ github.sha }}/otel-dotnet-auto-install.sh -O
+          sh ./otel-dotnet-auto-install.sh
+            test "$(ls -A "$HOME/.otel-dotnet-auto")"
+          curl -sSfL https://raw.githubusercontent.com/${{ github.repository }}/${{ github.sha }}/instrument.sh -O
+          . ./instrument.sh
+          ./test/test-applications/integrations/TestApplication.Smoke/bin/Release/net7.0/publish/TestApplication.Smoke
+            test "$(ls -A /var/log/opentelemetry/dotnet )"
+        '
```

Comment:
this would have to be triggered on `release` instead of `tag`, See: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

When we create a tag we do not have the release yet. We first create a tag an then we create a release.

Please test it on a fork.

Reply:
I'll go to the draft mode for the testing of this change